google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/project/ProjectSelector.java

```diff
@@ -0,0 +1,220 @@
+import com.google.cloud.tools.intellij.login.Services;
+import com.google.cloud.tools.intellij.login.ui.GoogleLoginIcons;
+import com.google.cloud.tools.intellij.util.GctBundle;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.intellij.openapi.ui.ComponentWithBrowseButton;
+import com.intellij.ui.HyperlinkLabel;
+import com.intellij.ui.components.JBLabel;
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.FlowLayout;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import javax.swing.Icon;
+import javax.swing.JPanel;
+import javax.swing.UIManager;
+import javax.swing.event.HyperlinkEvent.EventType;
+
+/**
+ * ProjectSelector allows the user to select a GCP project id. It shows selected project name and
+ * user account information. To change selection it uses {@link ProjectSelectionDialog} to call into
+ * {@link com.google.cloud.tools.intellij.login.IntegratedGoogleLoginService} to get the set of
+ * credentialed users and then into resource manager to get the set of projects.
+ */
+public class ProjectSelector extends JPanel {
+  static final int ACCOUNT_ICON_SIZE = 16;
+
+  private final List<ProjectSelectionListener> projectSelectionListeners = new ArrayList<>();
+
+  private HyperlinkLabelWithStateAccess projectNameLabel;
+  private HyperlinkLabelWithStateAccess accountInfoLabel;
+  private JBLabel projectAccountSeparatorLabel;
+
+  private CloudProject cloudProject;
+
+  public ProjectSelector() {
+    createUIComponents();
+    setSelectedProject(null);
+  }
+
+  /** @return project selection or null if no project is selected. */
+  public CloudProject getSelectedProject() {
+    return cloudProject;
+  }
+
+  /**
+   * Updates component UI and state with the new project and user account information.
+   *
+   * @param cloudProject New project/account information, null to clear selected project.
+   */
+  public void setSelectedProject(CloudProject cloudProject) {
+    this.cloudProject = cloudProject;
+    updateProjectAndUserInformation(cloudProject);
+  }
+
+  /**
+   * Adds a {@link ProjectSelectionListener} to this class's internal list of listeners. All
+   * ProjectSelectionListeners are notified when the selection is changed to a valid project.
+   *
+   * @param projectSelectionListener the listener to add
+   */
+  public void addProjectSelectionListener(ProjectSelectionListener projectSelectionListener) {
+    projectSelectionListeners.add(projectSelectionListener);
+  }
+
+  /**
+   * Removes a {@link ProjectSelectionListener} from this class's internal list of listeners.
+   *
+   * @param projectSelectionListener the listener to remove
+   */
+  public void removeProjectSelectionListener(ProjectSelectionListener projectSelectionListener) {
+    projectSelectionListeners.remove(projectSelectionListener);
+  }
+
+  private void createUIComponents() {
```

Comment:
the is the same name of the method IntelliJ auto creates for form-bound swing components when you choose "Custom Create":

![image](https://user-images.githubusercontent.com/1735744/33292725-3f4dce04-d398-11e7-8ef8-6799dc767748.png)

I'm not sure if its a problem really, but on first look I expected to see a form file for this class.

Reply:
Also, is there any reason you decided not to create a form for this?

Reply:
It's a very simple flow layout line with two hyperlinks, so it's easy to see how it's implemented in code, plus I was having issues with properly wrapping the components into IDEA standard ComponentWithBrowseButton<> widget using the form editor.